osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/infra/jms/core/OsgpCoreRequestMessageSender.java

```diff
@@ -30,11 +31,21 @@
 
     public void send(final RequestMessage requestMessage, final String messageType, final int messagePriority,
             final String ipAddress) {
-        this.send(requestMessage, messageType, messagePriority, ipAddress, null);
+        this.send(requestMessage, messageType, messagePriority, ipAddress, null, null);
     }
 
-    public void send(final RequestMessage requestMessage, final String messageType, final int messagePriority,
-            final String ipAddress, final Long scheduleTime) {
+    public void sendWithScheduledTime(final RequestMessage requestMessage, final String messageType,
+            final int messagePriority, final String ipAddress, final Long scheduledTime) {
+        this.send(requestMessage, messageType, messagePriority, ipAddress, scheduledTime, null);
+    }
+
+    public void sendWithDelay(final RequestMessage requestMessage, final String messageType, final int messagePriority,
+            final String ipAddress, final Long delay) {
+        this.send(requestMessage, messageType, messagePriority, ipAddress, null, delay);
+    }
+
+    private void send(final RequestMessage requestMessage, final String messageType, final int messagePriority,
+            final String ipAddress, final Long scheduledTime, final Long delay) {
 
         this.jmsTemplate.send(new MessageCreator() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 22 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1188)